extract-locales.js

```diff
@@ -0,0 +1,79 @@
+/**
+ *    Copyright (c) ppy Pty Ltd <contact@ppy.sh>.
+ *
+ *    This file is part of osu!web. osu!web is distributed with the hope of
+ *    attracting more community contributions to the core ecosystem of osu!.
+ *
+ *    osu!web is free software: you can redistribute it and/or modify
+ *    it under the terms of the Affero GNU General Public License version 3
+ *    as published by the Free Software Foundation.
+ *
+ *    osu!web is distributed WITHOUT ANY WARRANTY; without even the implied
+ *    warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *    See the GNU Affero General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Affero General Public License
+ *    along with osu!web.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const mkdirp = require('mkdirp');
+
+const buildPath = path.resolve(__dirname, 'resources/assets/build');
+const localesPath = path.resolve(buildPath, 'locales');
+const messagesPath = path.resolve(buildPath, 'messages.json');
+
+function extractLanguages() {
+  console.log('Extracting localizations...')
+  mkdirp.sync(localesPath);
```

Comment:
this should be in `writeTranslations` instead. And maybe followed by removing existing `.js` files as it might be filled with unrelated stuff.